### PR TITLE
fix: self contained styles followup bug fixes

### PR DESCRIPTION
**Pull request by jbadan**
## Related Issue

fixes https://github.com/SAP/fundamental-styles/issues/326

## Description
* moved Timepicker and Datepicker to "Patterns" as they no longer have their own CSS
* updated docs site examples for components that were overlooked in round 1
* updated scss files in components that had oversights
* update checkboxes and radio buttons to work
* removed menu_list child resets - this interfered with inheritance. 

POST PR FEEDBACK UPDATE:
added new `input-group__input` class for inputs and textarea children of input group to avoid specificity errors.

## Screenshots
Screen Shot 2019-09-12 at 5 04 46 PM

### CODE REVIEW

**Comment by bcullman** [comment]
perhaps add  to to calendar and time something like  

```
Looking for Datepicker? From a style perspective, a datepicker is simply a composition of our existing calendar, input and popup controls, but contains no new styles of its own. Therefore, we consider it a pattern. See the datepicker pattern here... 
```

in a yellow warning message box.

File: src/fonts.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+@import "../scss/fonts/72";
```
Comment (greg-a-smith) [inline src/fonts.scss]: Where is this used?

**Comment by greg-a-smith** [comment]
@bcullman I like that idea of calling out to patterns.  I'd do the same for Time Picker from the Time component.

File: src/fonts.scss (hunk 1 above)
Comment (jbadan) [reply src/fonts.scss]: We need it in fundamental react for the docs site at least.

File: scss/components/input-group.scss [hunk 2]
```diff
@@ -4,14 +4,21 @@
...
     @include fd-reset();
     display: flex;
 
+    > .#{$fd-namespace}-input {
+        @include fd-reset-child-spacing();
+        flex: 1 0 auto;
+        width: auto;
+        z-index: map-get($fd-z-index-levels, "first");
+    }
+
```
[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/input-group.scss]: It's the specificity issue we're running into in the implementation libraries - Screen Shot 2019-09-09 at 3 17 34 PM

`.fd-input` is not inheriting the * selector with enough specificity - so we're left with width: 100% on those and it's causing problems.

[the remark this answers is not in the document]
Comment (greg-a-smith) [reply scss/components/input-group.scss]: Crap!  So, in doing some research on this, although using the universal selector is valid CSS, it does _NOT_ provide _ANY_ specificity gains, which is why it was written this way in the first place.  Adding a block in `input-group.scss` with a class name of `fd-input` breaks the rules of self-contained styles so we need to find a better way to do this.

In doing some fiddling with this, I think we can achieve the results we were after by using the `fd-children` mixin.  Example:
```scss
    > * {
        @include fd-reset-child-spacing();
        @include fd-children() {
            flex: 1 0 auto;
            width: auto;
            z-index: map-get($fd-z-index-levels, "first");
        }
    }
```
This uses the `nth-child` pseudo selector to attach to any child elements which _DOES_ appear to make it more specific than just `fd-input`.

[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/input-group.scss]: Thanks for figuring that out! I've had a hell of a time trying to figure out what to do.

**Pull request merged**